src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.cs

```diff
@@ -554,13 +551,14 @@ internal void Run<TState>(string input, int startat, ref TState state, MatchCall
 
         /// <summary>Creates a new runner instance.</summary>
         private RegexRunner CreateRunner() =>
-            factory?.CreateInstance() ??
-            new RegexInterpreter(_code!, RegexParser.GetTargetCulture(roptions));
+            // The factory needs to be set by the ctor.  `factory` is a protected field, so it's possible a derived
+            // type nulls out the factory after we've set it, but that's the nature of the design.
+            factory!.CreateInstance();
 
         /// <summary>True if the <see cref="RegexOptions.Compiled"/> option was set.</summary>
         protected bool UseOptionC() => (roptions & RegexOptions.Compiled) != 0;
 
         /// <summary>True if the <see cref="RegexOptions.RightToLeft"/> option was set.</summary>
-        protected internal bool UseOptionR() => (roptions & RegexOptions.RightToLeft) != 0;
+        protected internal bool UseOptionR() => RightToLeft;
```

Comment:
I know it's not part of your PR, but should we just consider obsoleting this in the future?

Reply:
Yes, they're in the list in https://github.com/dotnet/runtime/issues/62573